Formula/s/search.rb

```diff
@@ -0,0 +1,18 @@
+class Search < Formula
+  include Language::Python::Virtualenv
+
+  desc "A command line tool to search words in files"
+  homepage "https://github.com/mark1n0/search"
+  url "https://github.com/mark1n0/search/archive/v0.1.tar.gz"
+  sha256 "74d52b69157f6dc4d8394f5144e72020b3cca41b89b21f54981d597b487f2199"
+
+  depends_on "python@3.9"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system bin/"search", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.